src/controllers/mainController.ts

```diff
@@ -865,9 +847,55 @@ export default class MainController implements vscode.Disposable {
      * Called by VS Code when user settings are changed
      * @param ConfigurationChangeEvent event that is fired when config is changed
      */
-    public onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): void {
+    public async onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): Promise<void> {
         if (e.affectsConfiguration(Constants.extensionName)) {
-            this._objectExplorerProvider.refresh(undefined);
+            let needsRefresh = false;
+            // user connections is a super set of object explorer connections
+            let userConnections: any[] = this._vscodeWrapper.getConfiguration(Constants.extensionName).get(Constants.connectionsArrayName);
+            let objectExplorerConnections = this._objectExplorerProvider.rootNodeConnections;
+
+            // if a connection(s) was/were manually removed
+            let staleConnections = objectExplorerConnections.filter((oeConn) => {
+                return !userConnections.some((userConn) => Utils.isSameConnection(oeConn, userConn));
+            });
+            // disconnect that/those connection(s) and then
+            // remove its/their credentials from the credential store
+            // and MRU
+            for (let conn of staleConnections) {
+                let profile = <IConnectionProfile>conn;
+                if (this.connectionManager.isActiveConnection(conn)) {
+                    const uri = this.connectionManager.getUriForConnection(conn);
+                    await this.connectionManager.disconnect(uri);
+                }
+                await this.connectionManager.connectionStore.removeRecentlyUsed(profile);
+                if (profile.authenticationType === Constants.sqlAuthentication &&
+                    profile.savePassword) {
+                        await this.connectionManager.deleteCredential(profile);
+                    }
+            }
+            // remove them from object explorer
+            await this._objectExplorerProvider.removeConnectionNodes(staleConnections);
+            needsRefresh = true;
```

Comment:
>needsRefresh [](start = 12, length = 12)

You're always refreshing now. I'd change this to `needsRefresh = staleConnections.count > 0`;